server/app/routes/admin.products.tsx

```diff
@@ -194,46 +264,66 @@ export default function Index() {
   );
 }
 
-function getProductBoxes(
-  products: SerializeFrom<ProductData>[],
-  submit: SubmitFunction
-) {
-  return products.map((product) => {
-    return (
-      <Card key={product.id} className="m-5 grow basis-72">
-        <Title>{product.name}</Title>
-        <div className="my-3">
-          <div>
-            <b>ID</b>: {product.id}
-          </div>
-          <div>
-            <b>Name</b>: {product.name}
-          </div>
-          <div>
-            <b>Reviews</b>: {product.reviewCount}
-          </div>
-          <div>
-            <b>Reports</b>: {product.reportCount}
-          </div>
-          <div>
-            <b>Created on</b>: {product.createdAt}
-          </div>
-          <div>
-            <b>Updated on</b>: {product.updatedAt}
-          </div>
+interface AdminProductCardProps {
+  product: SerializeFrom<ProductData>;
+  submit: SubmitFunction;
+  action: string;
+  setAction: (action: string) => void;
+  isSubmitting: boolean;
+}
+
+function AdminProductCard({
+  product,
+  submit,
+  action,
+  setAction,
+  isSubmitting,
+}: AdminProductCardProps) {
+  const navigation = useNavigation();
+  const isNavProdInfo =
+    navigation.state === "loading" &&
+    navigation.location.pathname === `/admin/product/${product.id}`;
+  return (
+    <Card className="m-5 grow basis-72">
+      <Title>{product.name}</Title>
+      <div className="my-3">
+        <div>
+          <b>ID</b>: {product.id}
+        </div>
+        <div>
+          <b>Name</b>: {product.name}
+        </div>
+        <div>
+          <b>Reviews</b>: {product.reviewCount}
+        </div>
+        <div>
+          <b>Reports</b>: {product.reportCount}
+        </div>
+        <div>
+          <b>Created on</b>: {product.createdAt}
         </div>
+        <div>
+          <b>Updated on</b>: {product.updatedAt}
+        </div>
+      </div>
 
-        <Link to={`/admin/product/${product.id}`}>
-          <Button className="mt-4 block" size="sm">
-            View Product Information
-          </Button>
+      <div className="flex flex-col gap-2">
+        <Link
+          to={`/admin/product/${product.id}`}
+          className={`${buttonVariants({ size: "sm" })} ${
```

Comment:
The 3 buttons in these product cards look too large when the last cards of a page are stretched to fill page width